main/src/view_model/conversation_details.vala

```diff
@@ -32,6 +33,7 @@ public class Dino.Ui.ViewModel.ConversationDetails : Object {
 
     public bool show_blocked { get; set; }
     public bool blocked { get; set; }
+    public bool domain_blocked { get; set; }
```

Comment:
Similarly to the NotificationSetting, you could create an enum that represents the blocking states: None, Blocked, DomainBlocked.

Reply:
Will do!

Reply:
My thought was to also use this enum for representing the state in the ViewModel and Model. So we'd have a `public BlockState block_state { get; set; }` that replaces `blocked` and `domain_blocked`.